tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,207 @@ def __init__(self, **kwargs):
+        epsilon = tf.math.abs(self.epsilon)
+        if self.use_eps_learned:
+            epsilon += self.eps_learned
+        nu2 = tf.reduce_mean(tf.square(inputs), axis=self.axis, keepdims=True)
+        inputs *= tf.math.rsqrt(nu2 + epsilon)
+        return self.gamma * inputs + self.beta
+
+    def _create_input_spec(self, input_shape):
+        ndims = len(tf.TensorShape(input_shape))
+        for idx, x in enumerate(self.axis):
+            if x < 0:
+                self.axis[idx] = ndims + x
+
+        # Validate axes
+        for x in self.axis:
+            if x < 0 or x >= ndims:
+                raise ValueError("Invalid axis: %d" % x)
+
+        if len(self.axis) != len(set(self.axis)):
+            raise ValueError("Duplicate axis: %s" % self.axis)
+
+        axis_to_dim = {x: input_shape[x] for x in self.axis}
+        self.input_spec = tf.keras.layers.InputSpec(ndim=ndims, axes=axis_to_dim)
+
+    def _check_axis(self):
+        if self.axis == 0:
+            raise ValueError(
+                "You are trying to normalize your batch axis. You may want to "
+                "use tf.layer.batch_normalization instead"
+            )
+        if self.axis == -1:
+            raise ValueError(
+                "You are trying to normalize your channel axis. You may want to "
+                "use GroupNormalization layer instead from tf_addons"
+            )
+        if self.axis != [1, 2]:
+            raise ValueError(
+                "FilterResponseNormalization operates on per-channel basis."
+                "Axis values should be spatial dimensions"
+            )
+
+    def _check_if_input_shape_is_none(self, input_shape):
+        dim1, dim2 = input_shape[self.axis[0]], input_shape[self.axis[1]]
+        if dim1 is None or dim2 is None:
+            raise ValueError(
+                "Axis " + str(self.axis) + " of "
+                "input tensor should have a defined dimension "
+                "but the layer received an input with shape " + str(input_shape) + "."
+            )
+
+    def _add_gamma_weight(self, input_shape):
+        # Get the channel dimension
+        dim = input_shape[-1]
+        shape = [1, 1, 1, dim]
+        # Initialize gamma with shape (1, 1, 1, C)
+        self.gamma = self.add_weight(
+            shape=shape,
+            name="gamma",
+            dtype=self.dtype,
+            initializer=self.gamma_initializer,
+            regularizer=self.gamma_regularizer,
+            constraint=self.gamma_constraint,
+        )
+
+    def _add_beta_weight(self, input_shape):
+        # Get the channel dimension
+        dim = input_shape[-1]
+        shape = [1, 1, 1, dim]
+        # Initialize beta with shape (1, 1, 1, C)
+        self.beta = self.add_weight(
+            shape=shape,
+            name="beta",
+            dtype=self.dtype,
+            initializer=self.beta_initializer,
+            regularizer=self.beta_regularizer,
+            constraint=self.beta_constraint,
+        )
```

Comment:
The 4D tensor shape of `gamma` and `beta` will only work for batched images. It won't work for output from FC. 

We have two ways forward:
1. Do an assert on `input_shape` to ensure it is 4D, and add a TODO to add support for more shapes in the future.
2. Modify the code to support various sizes. 

I'm in favour of 1 to reduce the scope of this PR.

Reply:
I am also in favor of 1 for now